dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
+        assertEquals(1, allRules.size());
+        final List<RuleAction> ruleActions = allRules.get(0).getRuleActions();
+
+        assertEquals(1, ruleActions.size());
+
+        assertEquals(ruleAction.getActionDefinition(), ruleAction.getActionDefinition());
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User without permission try to create a Rule's Action
+     * Should: Throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldNotSaveRuleAction() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User with permission try to delete a rule
+     * Should: Delete it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldDeleteRules() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        this.addPermission(role, host, true);
+
+        assertTrue(existRule(user, rule));
+
+        rulesAPI.deleteRule(rule, user, false);
+
+        assertFalse(existRule(user, rule));
+
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User without permission try to delete a rule
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldNotDeleteRules() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        try {
+            rulesAPI.deleteRule(rule, user, false);
+            throw new AssertionError("DotSecurityException Expected");
+        } catch (DotSecurityException e) {
+            assertFalse(existRule(user, rule));
+        }
+    }
+
+    private boolean existRule(User user, Rule rule) throws DotDataException, DotSecurityException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)